library/src/main/java/com/okta/oidc/AuthenticationPayload.java

```diff
@@ -71,6 +73,48 @@ public Builder setState(@Nullable String state) {
             return this;
         }
 
+        /**
+         * Identity provider (default is Okta, unless you are using Social Login or enterprise SAML)
+         * If using a social login provider use this to set the client id of the app.
+         *
+         * @param idp This is the client id of the app social login provider.
+         * @return current Builder
+         * @see "Social Login
+         * <https://developer.okta.com/docs/concepts/social-login/#features>"
+         */
+        public Builder setIdp(@Nullable String idp) {
+            if (!TextUtils.isEmpty(idp)) {
+                mAdditionalParameters.put(IDP, idp);
+            } else {
+                mAdditionalParameters.remove(IDP);
+            }
+            return this;
+        }
+
+        /**
+         * Identity provider (default is Okta, unless you are using Social Login or enterprise SAML)
+         * If using a social login provider use this to set the client id of the app.
+         *
+         * @param idpScope This is the client id of the app social login provider.
+         * @return current Builder
+         * @see "Social Login
+         * <https://developer.okta.com/docs/concepts/social-login/#features>"
+         */
+        public Builder setIdpScope(@Nullable String... idpScope) {
+            if (idpScope != null && idpScope.length > 0) {
+                StringBuilder scopes = new StringBuilder();
+                for (String scope : idpScope) {
+                    scopes.append(scope);
+                    scopes.append(" ");
```

Comment:
are scopes assured of not having spaces in them?

Reply:
scopes shouldn't have spaces in them per the [oidc spec](https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims), it is space delimited.